src/backend/core/api/viewsets.py

```diff
@@ -455,6 +447,81 @@ def create_for_owner(self, request):
             {"id": str(document.id)}, status=status.HTTP_201_CREATED
         )
 
+    @drf.decorators.action(
+        detail=True,
+        methods=["post"],
+        permission_classes=[permissions.IsAuthenticated, permissions.AccessPermission],
+        url_path="duplicate",
+    )
+    @transaction.atomic
+    def duplicate(self, request, *args, **kwargs):
+        """
+        Duplicate a document and store the links to attached files in the duplicated
+        document to allow cross-access.
+
+        Optionally duplicates accesses if `with_accesses` is set to true
+        in the payload.
```

Comment:
As it is now, `with_accesses` is not set from the payload but from the query string.
Why not using the payload btw ?

Reply:
No reason. Do you think it's better in the payload? I think you're right...